test/thread_factory.cpp

```diff
@@ -9,7 +12,92 @@ namespace {
 using namespace yaclib;
 // TODO(kononovk): #2 Add simple single/multi-threaded test
 
+void to_empty_factory(executor::IThreadFactoryPtr factory,
+                      size_t max_thread_count) {
+  auto do_nothing = [] {
+  };
+  for (size_t i = 0; i < max_thread_count; ++i) {
+    auto thread_ptr = factory->Acquire(do_nothing);
+    EXPECT_NE(thread_ptr, nullptr);
+    thread_ptr->Join();
```

Comment:
- [ ] you forgot to Release this
you should Acquire/Release for more complicated implementation